src/Lachain.Core/RPC/HTTP/FrontEnd/FrontEndService.cs

```diff
@@ -307,7 +308,7 @@ private string AddTxToPool(Transaction tx)
         {
             var wallet = _privateWallet.GetWalletInstance();
             if (wallet is null) return "0x0";
-            var receipt = _transactionSigner.Sign(tx, wallet.EcdsaKeyPair);
+            var receipt = _transactionSigner.Sign(tx, wallet.EcdsaKeyPair, HardforkHeights.IsHardfork_8Active(_stateManager.LastApprovedSnapshot.Blocks.GetTotalBlockHeight()));
```

Comment:
If this tx is later added to the pool, then we should use GetTotalBlockHeight() + 1

Reply:
Fixed